tests/containers/run_container_in_k8s.pm

```diff
@@ -22,7 +22,8 @@ sub run {
 
     my $cmd = '"cat", "/etc/os-release"';
 
-    my $provider = $self->provider_factory(service => 'EKS');
+    my $service = $self->get_k8s_service_name();
+    my $provider = $self->provider_factory(service => $service);
     $self->{provider} = $provider;
```

Comment:
Is there a reason why not use either `$provider` or `$self->{provider}` only?

Reply:
Good point, I am moving this part to the parent (k8sbasetest)

Reply:
Done